frontend/src/my-books/ShelfModal.tsx

```diff
@@ -0,0 +1,99 @@
+/*
+The book project lets a user keep track of different books they would like to read, are currently
+reading, have read or did not finish.
+Copyright (C) 2020  Karan Kumar
+
+This program is free software: you can redistribute it and/or modify it under the terms of the
+GNU General Public License as published by the Free Software Foundation, either version 3 of the
+License, or (at your option) any later version.
+
+This program is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+PURPOSE.  See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with this program.
+If not, see <https://www.gnu.org/licenses/>.
+*/
+
+import React, { Component } from "react";
+import Modal, { IModalProps } from "../shared/components/Modal";
+import Button from "@material-ui/core/Button";
+import TextField from "@material-ui/core/TextField";
+import "./ShelfModal.css";
+import Hidden from "@material-ui/core/Hidden";
+
+type MyState = { name: string };
+export default class ShelfModal extends Component<IModalProps, MyState> {
+  constructor(props: never) {
+    super(props);
+    this.state = { name: "" };
+    this.submitShelf = this.submitShelf.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+  }
+
+  handleChange = (event: any) => {
+    this.setState({ name: event.target.value });
+  };
+
+  submitShelf = (event: any) => {
+    alert(`Shelf added: ${this.state.name}`);
```

Comment:
Please remove the alert. If there is a ticket to hook this up to a backend, please leave a comment with the ticket number.